### PR TITLE
feat: use gh to release

### PR DESCRIPTION
**Pull request by Yu-Jack**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -35,14 +35,11 @@ jobs:
...
         if: startsWith(github.ref, 'refs/tags/')
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          prerelease: true
-          body_path: release.txt
-          files: ${{ steps.info.outputs.artifacts_dir }}/*
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release upload ${{ github.ref_name }} ${{ steps.info.outputs.artifacts_dir }}/*
```
Comment (bk201) [inline .github/workflows/release.yml:42]: Should we make it a prerelease to align with the original code?

Comment (Yu-Jack) [reply .github/workflows/release.yml:42]: I think we don't need to do it. It also upload tarball when we create a tag like this test https://github.com/Yu-Jack/docker-machine-driver-harvester/releases.

Comment (bk201) [reply .github/workflows/release.yml:42]: @FrankYang0529 do we need prerelease set to true?

Comment (FrankYang0529) [reply .github/workflows/release.yml:42]: Some of previous releases were created by github-actions. Probably, it's better to create a draft release first and manually change to latest release.

Screenshot 2024-12-18 at 4 03 11 PM

Comment (Yu-Jack) [reply .github/workflows/release.yml:43]: It depends on tag or release. If we prefer to create a tag first, then we can use draft release. If we prefer to create a release directly, then we don't need to create draft relase.

**Review by FrankYang0529 — APPROVED** [review]
LGTM. Thanks.

**Review by bk201 — APPROVED** [review]

**Comment by Yu-Jack** [comment]
don't approve it, i must fix it

**Pull request merged**